src/sagemaker/tensorflow/legacy_training.rst

```diff
@@ -0,0 +1,487 @@
+Preparing Legacy Mode training script
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+For Legacy Mode your TensorFlow training script must be a **Python 2.7** source file. The SageMaker TensorFlow
+docker image uses this script by calling specifically-named functions from this script.
+
+The training script **must contain** the following:
+
+- Exactly one of the following:
+
+  - ``model_fn``: defines the model that will be trained.
+  - ``keras_model_fn``: defines the ``tf.keras`` model that will be trained.
+  - ``estimator_fn``: defines the ``tf.estimator.Estimator`` that will train the model.
+
+- ``train_input_fn``: preprocess and load training data.
+- ``eval_input_fn``: preprocess and load evaluation data.
+
+In addition, it may optionally contain:
+
+- ``serving_input_fn``: Defines the features to be passed to the model during prediction. **Important:**
+    this function is used only during training, but is required to deploy the model resulting from training
+    to a SageMaker endpoint.
+
+Creating a ``model_fn``
+^^^^^^^^^^^^^^^^^^^^^^^
+
+A ``model_fn`` is a function that contains all the logic to support training, evaluation,
+and prediction. The basic skeleton for a ``model_fn`` looks like this:
+
+.. code:: python
+
+  def model_fn(features, labels, mode, hyperparameters):
+    # Logic to do the following:
+    # 1. Configure the model via TensorFlow operations
+    # 2. Define the loss function for training/evaluation
+    # 3. Define the training operation/optimizer
+    # 4. Generate predictions
+    # 5. Return predictions/loss/train_op/eval_metric_ops in EstimatorSpec object
```

Comment:
I think it'd be better to have these steps as part of the text rather than commented-out code